src/metrics/core.rs

```diff
@@ -0,0 +1,125 @@
+use std::sync::atomic::{AtomicBool, Ordering};
+
+use once_cell::sync::Lazy;
+use prometheus_client::{encoding::text::encode, registry::Registry};
+
+use crate::metrics::iroh;
+
+pub(crate) static CORE: Lazy<Core> = Lazy::new(Core::default);
+
+pub(crate) struct Core {
+    enabled: AtomicBool,
+    registry: Registry,
+    iroh_metrics: iroh::Metrics,
+}
+
+impl Default for Core {
+    fn default() -> Self {
+        let mut reg = Registry::default();
+        Core {
+            enabled: AtomicBool::new(false),
+            iroh_metrics: iroh::Metrics::new(&mut reg),
+            registry: reg,
+        }
+    }
+}
+
+impl Core {
+    pub(crate) fn registry(&self) -> &Registry {
+        &self.registry
+    }
+
+    pub(crate) fn iroh_metrics(&self) -> &iroh::Metrics {
+        &self.iroh_metrics
+    }
+
+    pub(crate) fn encode(&self) -> Result<Vec<u8>, std::io::Error> {
+        let mut buf = vec![];
+        encode(&mut buf, self.registry())?;
+        Ok(buf)
+    }
+
+    pub(crate) fn set_enabled(&self, enabled: bool) {
+        self.enabled.swap(enabled, Ordering::Relaxed);
+    }
+
+    pub(crate) fn is_enabled(&self) -> bool {
+        self.enabled.load(Ordering::Relaxed)
+    }
+}
+
+/// Defines the metric trait which provides a common interface for all value based metrics
+pub trait MetricType {
+    /// Returns the name of the metric
+    fn name(&self) -> &'static str;
+}
+
+/// Defines the histogram trait which provides a common interface for all  based metrics
+pub trait HistogramType {
+    /// Returns the name of the metric
+    fn name(&self) -> &'static str;
+}
+
+/// Definition of the base metrics collection interfaces.
+/// Instances imlementing the MetricsRecorder are expected to have a defined mapping between
```

Comment:
```suggestion
/// Definition of the base metrics collection interfaces.
///
/// Instances imlementing the MetricsRecorder are expected to have a defined mapping between
```

Leaving a blank line after a summary line renders much nicer in the indexes and search results in `cargo doc` output.  (Same in other doc comments)